homeassistant/components/remote/xiaomi_miio.py

```diff
@@ -0,0 +1,231 @@
+
+    # Create handler
+    _LOGGER.info("Initializing with host %s (token %s...)", host, token[:5])
+    device = ChuangmiIr(host, token)
+
+    try:
+        device.info()
+
+    # This should be DeviceError but python-miio returns wrong except.
+    except ChecksumError as ex:
+        _LOGGER.error("Token not accepted by device : %s", ex)
+        return
+
+    if PLATFORM not in hass.data:
+        hass.data[PLATFORM] = {}
+
+    entity_id = config.get(CONF_NAME, "xiaomi_miio_" +
+                           host.replace('.', '_'))
+    slot = config.get(CONF_SLOT)
+    timeout = config.get(CONF_TIMEOUT)
+
+    hidden = config.get(ATTR_HIDDEN)
+
+    xiaomi_miio_remote = XiaomiMiioRemote(
+        entity_id, device, slot, timeout, hidden, config.get(CONF_COMMANDS))
+
+    hass.data[PLATFORM][
+        entity_id.replace(' ', '_').lower()] = xiaomi_miio_remote
+
+    async_add_devices([xiaomi_miio_remote])
+
+    @asyncio.coroutine
+    def _learn_command(call):
+        """Handle a learn command."""
+        entity_id = call.data.get('entity_id').split('.')[1]
+
+        if entity_id not in hass.data[PLATFORM]:
+            _LOGGER.error("entity_id: '%s' not found", entity_id)
+            return
+
+        entity = hass.data[PLATFORM][entity_id]
+
+        device = entity.device
+
+        slot = call.data.get(CONF_SLOT, entity.slot)
+
+        yield from hass.async_add_job(device.learn, slot)
+
+        timeout = call.data.get(CONF_TIMEOUT, entity.timeout)
+
+        _LOGGER.info("Press the key you want Home Assistant to learn")
+        start_time = utcnow()
+        while (utcnow() - start_time) < timedelta(seconds=timeout):
+            command = yield from hass.async_add_job(
+                device.read, slot)
+            if command['code']:
+                log_msg = "Received command is: {}".\
+                          format(command['code'])
+                _LOGGER.info(log_msg)
+                hass.components.persistent_notification.async_create(
+                    log_msg, title='Xiaomi Miio Remote')
+                return
+            yield from asyncio.sleep(1, loop=hass.loop)
+        _LOGGER.error("Timeout. No infrared command captured")
+        hass.components.persistent_notification.async_create(
+            "Timeout. No infrared command captured",
+            title='Xiaomi Miio Remote')
+
+    hass.services.async_register(remote.DOMAIN, SERVICE_LEARN, _learn_command,
+                                 schema=LEARN_COMMAND_SCHEMA)
+
+
+class XiaomiMiioRemote(Entity):
+    """Representation of a Xiaomi Miio Remote device."""
+
+    def __init__(self, friendly_name, device,
+                 slot, timeout, hidden, commands):
+        """Initialize the remote."""
+        self._name = friendly_name
+        self._device = device
+        self._is_hidden = hidden
+        self._slot = slot
+        self._timeout = timeout
+        self._state = False
+        self._commands = commands
+
```

Comment:
Please add a should_poll method. Just to make clear it's a polling entity or not.

Reply:
done